0000-deb-repo-sync-gpg-check.md

```diff
@@ -0,0 +1,178 @@
+
+The field must be shown as a selector. The values used to populate the selector come from the existing table `rhnCryptoKey`. Only keys for current organization must be shown.
+
+An additional option `Default keys` must shown in the selector as default. If this option is selected, the validation will be done using the GPG keys from package `uyuni-build-keys`.
+
+When saving the repository data, in case there was a GPG key selected other then `Default keys` the assignment will be saved into the new table `rhnContentSourceGpg`.
+
+### Key usage in the API
+
+The API method `channel.software.create_repo()` must be extended with one optional arguments to allow specifying the `GPG key ID`.
+
+An error should be thrown if the the GPG key does not exist.
+
+### Key import via `spacewalk-repo-sync`
+
+`spacewalk-repo-sync` must be extended to be able to import the GPG keys configured in `rhnContentSourceGpg`. Similarly to Yum repos, the user should be asked for confirmation when importing the keys.
+
+### Metadata and signature lookup
+
+In order to verify integrity of the `Packages.gz` file Uyuni must first lookup the `Release` file and verify its signature. This file can be in one of these locations:
+- `http://mirror.example.com/$ARCHIVE_ROOT/dists/$DIST` in case of repos with a `dists` directory when the Uyuni repo URL points to `http://mirror.example.com/$ARCHIVE_ROOT/dists/$DIST/$COMPONENT/binary-$ARCH/`.
+- `http://mirror.example.com/$REPODIR/` in the case of flat repositories
+
+Looking up the `Release` file should be done automatically when syncing of the repo. An error should be shown if it's not possible to locate the `Release` file and its corresponding signature.
+
+### Metadata Verification
+
+Uyuni should verify the integrity of the `Packages.gz|xz` files using the checksums present in the `Release` file.
+
+### Signature verification
+
+If the `InRelease` or `Release.pgg` exist in the repo to be synced then Uyuni must verify the `Release` file. If the repo has signed metadata enabled but no GPG key selected, the default keyring must be used:
+```
+gpgv --keyring /var/lib/spacewalk/gpgdir/pubring.gpg InRelease
+```
+or
+```
+gpgv --keyring /var/lib/spacewalk/gpgdir/pubring.gpg Release.gpg Release
+```
+
+Otherwise the custom GPG key associated with the repo must be used:
+```
+gpgv --keyring </path/to/temp/keyring/file.gpg> InRelease
+```
+or
+```
+gpgv --keyring </path/to/temp/keyring/file.gpg> Release.gpg Release
+```
+
+The signature verification should be done during repo syncing when the metadata is parsed.
+Verification must be done only if the user has configured the GPG signing of the repos to be synced. Otherwise a warning must be printed in the logs about the lack of GPG configuration.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+No drawbacks.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Support only a very limited set of additional fields like `Multi-Arch` while discarding all others.
```

Comment:
this is an alternative for a different RFC.

Reply:
Removed.